crates/cast/src/starknet_commands/script/init.rs

```diff
@@ -0,0 +1,140 @@
+use anyhow::{anyhow, Context, Ok, Result};
+use camino::Utf8PathBuf;
+use std::fs;
+
+use cast::helpers::constants::SCRIPTS_DIR;
+use cast::helpers::response_structs::ScriptInitResponse;
+use cast::helpers::scarb_utils::get_cairo_version;
+use clap::Args;
+use indoc::{formatdoc, indoc};
+use scarb_metadata::ScarbCommand;
+
+#[derive(Args, Debug)]
+pub struct Init {
+    /// Name of a script to create
+    pub script_name: String,
+}
+
+pub fn init(init_args: &Init) -> Result<ScriptInitResponse> {
+    let script_root_dir_path = get_script_root_dir_path(&init_args.script_name)?;
+
+    init_scarb_project(&init_args.script_name, &script_root_dir_path)?;
+    add_dependencies(&script_root_dir_path)?;
+    modify_files_in_src_dir(&init_args.script_name, &script_root_dir_path)?;
+
+    Ok(ScriptInitResponse {
+        status: format!("Successfully initialized `{}`", init_args.script_name),
```

Comment:
We should consider unifying status messages between run and init (introduce a `ScriptStatus` enum for example). Not necessary in this PR.

Reply:
if not in this PR, let's create an issue and link it here :+1:

Reply:
https://github.com/foundry-rs/starknet-foundry/issues/1408